main/src/library/Condition.flix

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2021 Jonathan Lindegaard Starup
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+///
+/// A wrapper around a Java Condition.
+///
+opaque type Condition = ##java.util.concurrent.locks.Condition
+
+pub enum ConditionError {
+    case MonitorStateError(##java.lang.IllegalMonitorStateException)
```

Comment:
OK, I see what you mean.

What about we just ignore these exceptions entirely. They are not really meant to be caught anyway. 

Like, I would just drop all the try blocks.

Reply:
I don't think IllegalMonitorStateException can even happen! Only if you have illegal bytecode.

Reply:
OK, new suggestion:

- Drop `MonitorStateError` and keep everything else as-is. I think that is reasonable. What do you think?

Reply:
OK OK. I guess I am confused. Must be late.

I think the current design is right!